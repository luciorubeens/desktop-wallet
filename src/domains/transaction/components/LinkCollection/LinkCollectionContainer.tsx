import { Collapse } from "app/components/Collapse";
import { Icon } from "app/components/Icon";
import React, { useState } from "react";

const CollapseToggleIcon = ({ isExpanded }: { isExpanded?: boolean }) =>
	isExpanded ? (
		<Icon
			name="ChevronUp"
			width={10}
			height={10}
			className="flex items-center justify-center w-5 h-5 text-white rounded-full bg-theme-primary"
		/>
	) : (
		<Icon
			name="ChevronDown"
			width={10}
			height={10}
			className="flex items-center justify-center w-5 h-5 rounded-full text-theme-primary bg-theme-primary-contrast"
		/>
	);

type Props = {
	defaultIsExpanded?: false;
	title: string;
	description: string;
	children: React.ReactNode;
};

export const LinkCollectionContainer = ({ defaultIsExpanded, title, description, children }: Props) => {
	const [isExpanded, setIsExpanded] = useState<boolean>(defaultIsExpanded!);

	return (
		<div data-testid="LinkCollectionContainer">
			<div
				data-testid="LinkCollectionContainer__header"
				className="flex items-center justify-between cursor-pointer"
				onClick={() => setIsExpanded((prev) => !prev)}
			>
				<span className="text-lg font-semibold">{title}</span>

				<div data-testid="LinkCollectionContainer__collapse-toggle">
					<CollapseToggleIcon isExpanded={isExpanded} />
				</div>
			</div>

			<p data-testid="LinkCollectionContainer__description" className="mt-2 text-theme-neutral-dark">
				{description}
			</p>

			<Collapse data-testid="LinkCollectionContainer__collapse" isOpen={isExpanded}>
				<div className="mt-4">{children}</div>
			</Collapse>
		</div>
	);
};
