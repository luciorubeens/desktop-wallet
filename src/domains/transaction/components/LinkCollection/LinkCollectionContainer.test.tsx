import React from "react";
import { act, fireEvent, render } from "utils/testing-library";

import { LinkCollectionContainer } from "./LinkCollectionContainer";

describe("LinkCollectionContainer", () => {
	const title = "Test Title";
	const description = "Test Description";
	const Component = () => (
		<LinkCollectionContainer title={title} description={description}>
			<p>Test</p>
		</LinkCollectionContainer>
	);

	it("should render children", () => {
		const { getByText, asFragment } = render(<Component />);
		expect(getByText("Test")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should show title and description", () => {
		const { getByText } = render(<Component />);
		expect(getByText(title)).toBeInTheDocument();
		expect(getByText(description)).toBeInTheDocument();
	});

	it("should toggle collapse", () => {
		const { getByTestId } = render(<Component />);
		const header = getByTestId("LinkCollectionContainer__header");
		const collapse = getByTestId("LinkCollectionContainer__collapse");

		expect(collapse).toHaveAttribute("aria-hidden", "true");

		act(() => {
			fireEvent.click(header);
		});

		expect(collapse).toHaveAttribute("aria-hidden", "false");
	});
});
