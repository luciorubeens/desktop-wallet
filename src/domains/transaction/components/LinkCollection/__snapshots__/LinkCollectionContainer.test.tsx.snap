// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCollectionContainer should render children 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionContainer"
  >
    <div
      class="flex items-center justify-between cursor-pointer"
      data-testid="LinkCollectionContainer__header"
    >
      <span
        class="text-lg font-semibold"
      >
        Test Title
      </span>
      <div
        data-testid="LinkCollectionContainer__collapse-toggle"
      >
        <div
          class="sc-AxjAm iepfoc flex items-center justify-center w-5 h-5 rounded-full text-theme-primary bg-theme-primary-contrast"
          height="10"
          width="10"
        >
          <svg>
            chevron-down.svg
          </svg>
        </div>
      </div>
    </div>
    <p
      class="mt-2 text-theme-neutral-dark"
      data-testid="LinkCollectionContainer__description"
    >
      Test Description
    </p>
    <div
      aria-hidden="true"
      data-testid="LinkCollectionContainer__collapse"
      style="height: 0px; max-height: auto; opacity: 0; overflow: hidden;"
    >
      <div
        class="mt-4"
      >
        <p>
          Test
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
